### PR TITLE
EWPP-4233: Making editorial logic generic to all entity types.

### PR DESCRIPTION
**Pull request by upchuk**

### CODE REVIEW

File: modules/oe_editorial_corporate_workflow/oe_editorial_corporate_workflow.module [hunk 1]
```diff
@@ -8,66 +8,79 @@
...
  *
- * Force the Draft state on node forms.
+ * Force the Draft state on corporate workflow forms.
  */
-function oe_editorial_corporate_workflow_form_node_form_alter(&$form, FormStateInterface $form_state, $form_id) {
-  /** @var \Drupal\node\NodeInterface $node */
-  $entity = $form_state->getFormObject()->getEntity();
-  if (!$entity instanceof NodeInterface) {
+function oe_editorial_corporate_workflow_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $form_object = $form_state->getBuildInfo()['callback_object'];
```
Comment (nagyad) [inline modules/oe_editorial_corporate_workflow/oe_editorial_corporate_workflow.module]: You can use: `$form_state->getFormObject()`

File: modules/oe_editorial_entity_version/src/Form/CorporateWorkflowEntityRevisionRevertForm.php [hunk 2]
```diff
@@ -127,46 +128,42 @@ public function getConfirmText() {
...
    */
-  public function buildForm(array $form, FormStateInterface $form_state, $node_revision = NULL) {
-    // We need to anticipate string or NodeInterface types of $node_revision to
-    // be compatible with both core versions v9.2 and v9.3.
-    // @see: https://www.drupal.org/project/drupal/issues/2730631
-    $this->revision = $node_revision;
-    if (!$this->revision instanceof NodeInterface) {
-      $this->revision = $this->entityTypeManager->getStorage('node')->loadRevision($node_revision);
-    }
+  public function buildForm(array $form, FormStateInterface $form_state, $_entity_revision = NULL) {
```
Comment (nagyad) [inline modules/oe_editorial_entity_version/src/Form/CorporateWorkflowEntityRevisionRevertForm.php:131]: $entity_revision

File: modules/oe_editorial_corporate_workflow/oe_editorial_corporate_workflow.module [hunk 3]
```diff
@@ -8,66 +8,79 @@
...
+    $route_entity = \Drupal::routeMatch()->getParameter($entity_type);
+    if (!$route_entity instanceof ContentEntityInterface) {
+      continue;
+    }
+
+    if (!in_array($route_entity->bundle(), $entity_types[$entity_type])) {
+      continue;
+    }
+
+    $entity = $route_entity;
```
Comment (nagyad) [inline modules/oe_editorial_corporate_workflow/oe_editorial_corporate_workflow.module:35]: We can stop the cycle here, no need to iterate through all once we found the entity in the parameter.

**Review by nagyad — CHANGES_REQUESTED** [review]

File: modules/oe_editorial_corporate_workflow/tests/src/Traits/CorporateWorkflowTrait.php [hunk 4]
```diff
@@ -23,17 +24,17 @@ protected function getEntityTypeManager(): EntityTypeManagerInterface {
...
+   * Sends the entity through the moderation states to reach the target.
    *
-   * @param \Drupal\node\NodeInterface $node
-   *   The node.
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The entity.
    * @param string $target_state
    *   The target moderation state,.
    *
    * @return \Drupal\node\NodeInterface
```
Comment (nagyad) [inline modules/oe_editorial_corporate_workflow/tests/src/Traits/CorporateWorkflowTrait.php]: This should be changed too.

File: modules/oe_editorial_corporate_workflow/tests/src/Traits/CorporateWorkflowTrait.php [hunk 5]
```diff
@@ -42,23 +43,38 @@ protected function moderateNode(NodeInterface $node, string $target_state): Node
...
-      return $node;
+      return $entity;
     }
 
     $pos = array_search($current_state, $states);
     foreach (array_slice($states, $pos + 1) as $new_state) {
-      $node = $revision ?? $node;
-      $revision = $this->getEntityTypeManager()->getStorage('node')->createRevision($node);
+      $entity = $revision ?? $entity;
+      $revision = $this->getEntityTypeManager()->getStorage('node')->createRevision($entity);
```
Comment (nagyad) [inline modules/oe_editorial_corporate_workflow/tests/src/Traits/CorporateWorkflowTrait.php]: The getStorage() should be based on the entity type. This won't work for other entities.

**Review by nagyad — COMMENTED** [review]
Would be nice to address the CorporateWorkflowInstaller class too.

File: modules/oe_editorial_entity_version/src/Form/CorporateWorkflowEntityRevisionRevertForm.php [hunk 6]
```diff
@@ -97,7 +98,7 @@ public static function create(ContainerInterface $container) {
    * {@inheritdoc}
    */
   public function getFormId() {
-    return 'node_version_restore_confirm';
+    return 'entity_version_restore_confirm';
```
Comment (nagyad) [inline modules/oe_editorial_entity_version/src/Form/CorporateWorkflowEntityRevisionRevertForm.php:101]: We should have a test covering link lists similar to NodeVersionRestoreTest.

File: modules/oe_editorial_entity_version/src/Form/CorporateWorkflowEntityRevisionRevertForm.php (hunk 2 above)
Comment (upchuk) [reply modules/oe_editorial_entity_version/src/Form/CorporateWorkflowEntityRevisionRevertForm.php:131]: No, this is the param name that is being passed.

File: modules/oe_editorial_entity_version/src/Form/CorporateWorkflowEntityRevisionRevertForm.php (hunk 6 above)
Comment (upchuk) [reply modules/oe_editorial_entity_version/src/Form/CorporateWorkflowEntityRevisionRevertForm.php:101]: I think it's ok, the functionality is tested with an entity type

**Comment by upchuk** [comment]
> Would be nice to address the CorporateWorkflowInstaller class too.

It's ok. This one is used primarily for nodes and the usage is mostly for adding new bundles. it's very rare other entity types are gonna get the editorial flow.

**Review by 22Alexandra — DISMISSED** [review]

**Review by nagyad — DISMISSED** [review]

**Review by 22Alexandra — APPROVED** [review]

**Pull request merged**